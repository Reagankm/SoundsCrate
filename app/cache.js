/* @flow */

export default {
  update: updateCache,
  get: getCache,
};

const cache = {};

function updateCache(folderId: string, pageNumber: number, records: Array<string>) {
  if (!cache[folderId]) {
    cache[folderId] = [];
  }
  cache[folderId].push({ pageNumber, records });
}

function getCache(folderId: string, pageNumber: number): { pageNumber: number, records: Array<string> } {
  return cache[folderId] && cache[folderId].find(page => page.pageNumber === pageNumber);
};
