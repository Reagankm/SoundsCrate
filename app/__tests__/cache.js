/* @flow */
import cache from '../cache';

describe('app/cache', () => {
  const entry = {
    pageNumber: 1,
    records: ['foo', 'bar', 'baz'],
  };
  const secondPageEntry = {
    pageNumber: 2,
    records: ['some', 'cached', 'data'],
  }
  it('should return undefined if record is not found', () => {
    expect(cache.get(1, 1)).toBe(undefined);
  });
  it('should update the cache with an entry', () => {
    cache.update({
      folderId: 1,
      entry,
      totalPages: 2,
    });
    expect(cache.get(1, 1)).toMatchSnapshot();
  });
  it('should cache by page', () => {
    cache.update({
      folderId: 1,
      entry: secondPageEntry,
      totalPages: 2,
    });
    expect(cache.get(1, 1)).toMatchSnapshot();
    expect(cache.get(1, 2)).toMatchSnapshot();
  });
});
