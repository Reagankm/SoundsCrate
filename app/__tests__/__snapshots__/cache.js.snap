// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/cache should cache by page 1`] = `
Object {
  "records": Array [
    "foo",
    "bar",
    "baz",
  ],
  "totalPages": 2,
}
`;

exports[`app/cache should cache by page 2`] = `
Object {
  "records": Array [
    "some",
    "cached",
    "data",
  ],
  "totalPages": 2,
}
`;

exports[`app/cache should update the cache with an entry 1`] = `
Object {
  "records": Array [
    "foo",
    "bar",
    "baz",
  ],
  "totalPages": 2,
}
`;
